pinot-query-runtime/src/main/java/org/apache/pinot/query/service/dispatch/QueryDispatcher.java

```diff
@@ -146,18 +156,39 @@ public QueryResult submitAndReduce(RequestContext context, DispatchableSubPlan d
     Set<QueryServerInstance> servers = new HashSet<>();
     try {
       submit(requestId, dispatchableSubPlan, timeoutMs, servers, queryOptions);
-      try {
-        return runReducer(requestId, dispatchableSubPlan, timeoutMs, queryOptions, _mailboxService);
-      } finally {
-        if (isQueryCancellationEnabled()) {
-          _serversByQuery.remove(requestId);
-        }
-      }
+      return runReducer(requestId, dispatchableSubPlan, timeoutMs, queryOptions, _mailboxService);
+    } catch (Exception ex) {
```

Comment:
This `try-catch(Exception)-catch(Throwable)` chain feels a bit weird, I would at least add some comment explaining who is catching what.

Reply:
Why? All throwables that are not exceptions (basically Errors) are going to be cancelled without trying to recover. Normally I wouldn't catch Throwable, but it was already caught in the original code.